images/retry-trigger-jobs/retry_trigger_jobs.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+import json
+import os
+import urllib.parse
+from datetime import datetime, timedelta, timezone
+from requests import Session
+from requests.adapters import HTTPAdapter, Retry
+
+import sentry_sdk
+
+
+sentry_sdk.init(traces_sample_rate=1.0)
+
+GITLAB_API_URL = "https://gitlab.spack.io/api/v4/projects/2"
+
+
+def paginate(session, query_url):
+    """Helper method to get all pages of paginated query results"""
+    results = []
+
+    while query_url:
+        resp = session.get(query_url)
+
+        if resp.status_code == 401:
+            print(" !!! Unauthorized to make request, check GITLAB_TOKEN !!!")
+            return []
```

Comment:
Since we're instrumenting this script with Sentry, we can let this raise an exception and that will notify us if something is wrong instead of silently failing.
```suggestion
        resp.raise_for_status()
```